src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +427,69 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
```

Comment:
For the case above there is still a bug, thank you @chinpei215 will investigate.

This is the current (buggy) result:
### TeamsTable (same for WorkersTable)
```php
        $this->primaryKey('id');

        // GOOD PART
        $this->belongsToMany('Workers', [
            'foreignKey' => ['team_id'],
            'targetForeignKey' => ['worker_id'],
            'joinTable' => 'teams_workers'
        ]);
        // BUGGY PART
        $this->belongsToMany('Workers', [
            'foreignKey' => ['team_id'],
            'targetForeignKey' => ['position_id'],
            'joinTable' => 'teams_workers'
        ]);
    }
```